SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenFrames.swift

```diff
@@ -0,0 +1,94 @@
+//
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import QuartzCore
+import UIKit
+
+class ScreenFrames: NSObject {
+
+    private var isRunning = false
+    private var displayLink: CADisplayLink?
+    /// Anything less than 59 FPS is slow.
+    private var slowFrameThreshold: CFTimeInterval = 16
+    private var frozenFrameThreshold: CFTimeInterval = 700
+
+    private var slowCount: Int = 0
+    private var frozenCount: Int = 0
+    private var startedTime: CFTimeInterval = CACurrentMediaTime()
+
+    override init() {
+            super.init()
+            isRunning = false
+    }
+
+    func startTracking() {
+        isRunning = true
+        stopTracking() /// make sure to stop a previous running display link
+        let displayLink = CADisplayLink(target: self, selector: #selector(displayLinkCallback))
+        displayLink.preferredFramesPerSecond = SplunkRum.configuredOptions?.framesPerSecond ?? 0  /// (optional) if you do not define , then device uses maxFramesPS
```

Comment:
I think we can always have it run at max frames per second (the default behaviour)

Reply:
That's an optional feature I added for testing.

Reply:
But why do we have it in the final code?

Reply:
removed optional